src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingToAndFromManyAddressesSpecification_Steps.cs

```diff
@@ -0,0 +1,189 @@
+﻿using Stratis.Bitcoin.IntegrationTests.EnvironmentMockUpHelpers;
+using Stratis.Bitcoin.IntegrationTests.TestFramework;
+using Xunit.Abstractions;
+
+namespace Stratis.Bitcoin.IntegrationTests.Wallet
+{
+    using Stratis.Bitcoin.Features.Consensus;
+    using Stratis.Bitcoin.Features.Wallet;
+    using NBitcoin;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Stratis.Bitcoin.Features.Wallet.Controllers;
+    using Stratis.Bitcoin.Features.Wallet.Models;
+    using FluentAssertions;
+    using Stratis.Bitcoin.IntegrationTests.Builders;
+
+    public partial class SendingToAndFromManyAddressesSpecification : BddSpecification
+    {
+        private SharedSteps sharedSteps;
+        private const string WalletName = "mywallet";
+        private const string WalletPassword = "123456";
+        private const string WalletAccountName = "account 0";
+        private IDictionary<string, CoreNode> nodes;
+        private NodeGroupBuilder nodeGroupBuilder;
+        private const string NodeOne = "one";
+        private const string NodeTwo = "two";
+        private long stratisReceiverTotalAmount;
+        private TransactionBuildContext transactionBuildContext;
+        private int CoinBaseMaturity;
+
+        // Output helper allows test step names to be logged.
+        public SendingToAndFromManyAddressesSpecification(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+        protected override void BeforeTest()
+        {
+            this.sharedSteps = new SharedSteps();
+            this.nodeGroupBuilder = new NodeGroupBuilder();
+        }
+
+        protected override void AfterTest()
+        {
+            this.nodeGroupBuilder.Dispose();
+        }
+
+        private void two_connected_nodes()
+        {
+            this.nodes = this.nodeGroupBuilder
+                .StratisPowNode(NodeOne).Start().NotInIBD().WithWallet(WalletName, WalletPassword)
+                .StratisPowNode(NodeTwo).Start().NotInIBD().WithWallet(WalletName, WalletPassword)
+                .WithConnections()
+                    .Connect(NodeOne, NodeTwo)
+                    .AndNoMoreConnections()
+                .Build();
+        }
+
+        private void node1_sends_funds_to_node2_TO_fifty_addresses()
+        {
+            // Mine block
+            this.CoinBaseMaturity = (int)this.nodes[NodeOne].FullNode.Network.Consensus.Option<PowConsensusOptions>().CoinbaseMaturity;
+
+            this.sharedSteps.MineBlocks(this.CoinBaseMaturity + 50
```

Comment:
why +50 out of curiousity?

Reply:
Need a large amount to cover sending from 50 addresses.  I got to that figure using trial and error.